Servers/Administration/OJS.Servers.Administration/Controllers/ContestCategoriesController.cs

```diff
@@ -134,10 +142,10 @@ protected override Task BeforeEntitySaveOnCreateAsync(ContestCategory entity, Ad
 
     private async Task CascadeDeleteCategories(ContestCategory contestCategory)
     {
-        foreach (var children in contestCategory.Children.ToList())
+        await contestCategory.Children.ForEachAsync(async children =>
         {
             await this.CascadeDeleteCategories(children);
-        }
+        });
```

Comment:
Delete in async foreach is dangerous. Same db context instance cannot be used for parallel updates.

Reply:
With the proper includes method works just fine without it being async. The changes were probably made in the testing of problems with AutoCrudAdmin and are no longer needed so i reverted them